### ISSUE TITLE
Overwriting subject in CSR when signing it

### ISSUE THREAD

**Issue by weisinc** [cloudflare/cfssl#753]
I am trying to sign a CSR and overwrite its subject when the cert is generated but I kept getting the error no matter how I tried to format the curl string.
{"success":false,"result":null,"errors":[{"code":400,"message":"Unable to parse sign request"}],"messages":[]}

Here is my curl command string:
curl -d '{ "certificate_request":"-----BEGIN CERTIFICATE REQUEST-----<certificate>-----END CERTIFICATE REQUEST-----", "hosts":["www.example.com"], "subject":[{"C":"US", "ST":"California", "L":"San Francisco", "O":"example.com"}], "CN": "www.example.com"}'  http://$CFSSLHOST:8888/api/v1/cfssl/sign

Have spent many hours on it. Hope really appreciate it if someone can tell me the obvious. I am using CFSSL 1.2.0.  Thanks ahead.

**Comment by lziest** [comment]
where did you get the json object? The "subject" object is not structured to the expected one, and "CN" is not accepted.

See subject structure here https://github.com/cloudflare/cfssl/blob/master/signer/signer.go#L29

**Comment by weisinc** [comment]
I don't know the structure of the subject. I simply added it to the example I found in /doc/api/endpoint_sign.txt and basically doing trail and error on it. Thanks for sharing the struct.
I tried it with:
"subject": "{"CN": "www.example.com", "names": [{"C":"US", "ST":"California", "L":"San Francisco", "O":"Cisco, Inc."}]"
It is still giving error. Is SerialNumber mandatory in the Subject struct? What would be right structure in my example? Why is the "CN" not acceptable?
